atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPool.java

```diff
@@ -371,28 +373,35 @@ private boolean tableAlreadyExists(Client client, String caseInsensitiveTableNam
 
     private void refreshTokenRanges() {
         try {
+            ImmutableRangeMap.Builder<LightweightOPPToken, List<InetSocketAddress>> newTokenRing = ImmutableRangeMap.builder();
+
+            // grab latest token ring view from a random node in the cluster
             List<TokenRange> tokenRanges = getRandomGoodHost().runWithPooledResource(describeRing);
 
-            ImmutableRangeMap.Builder<LightweightOPPToken, List<InetSocketAddress>> newTokenRing = ImmutableRangeMap.builder();
-            for (TokenRange tokenRange : tokenRanges) {
-                List<InetSocketAddress> hosts = Lists.transform(tokenRange.getEndpoints(), new Function<String, InetSocketAddress>() {
-                    @Override
-                    public InetSocketAddress apply(String endpoint) {
+            if (tokenRanges.size() == 1) { // RangeMap needs a little help with weird 1-node, 1-vnode, this-entire-feature-is-useless case
+                String onlyEndpoint = Iterables.getOnlyElement(Iterables.getOnlyElement(tokenRanges).getEndpoints());
```

Comment:
Just because Tanya was also confused and asked about it, I didn't want to get into it just to be an ass before:

The way this is worded probably means you're thinking about it wrong; this is all way underneath the concept of data having RF.
This should be thought of more like having primary ownership for a range, and there's an entirely separate and user-configurable codepath for determining replica placement, which may or may not be based on the tokens here or other external data (i.e. in practice extra metadata from somewhere else about what boxes are where and what boxes are in what datacenters)

The only time you should have two different nodes with the same tokens is during an old-school ES-style token migration (with one of the nodes in the state MIGRATING) rather than doing a more cassandra-y way of adding new boxes, which is token range bisection with new randomly generated tokens and a repair operation.